crates/ecs/src/lib.rs

```diff
@@ -616,13 +622,18 @@ pub struct World {
     /// This will result in two vectors containing the indices of the `Archetype`s that store these types.
     /// By taking the intersection of these vectors you will know which `Archetype` contain both A and B.
     /// This could be the archetypes: (A,B), (A,B,C), (A,B,...) etc.
-    component_typeid_to_archetype_indices: HashMap<TypeId, HashSet<ArchetypeIndex>>,
+    component_typeid_to_archetype_indices:
+        FnvHashMap<TypeId, HashSet<ArchetypeIndex, BuildHasherDefault<NoHashHasher<usize>>>>,
```

Comment:
How come you're using both `FnvHashMap` and `BuildHasherDefault<NoHashHasher<usize>>`? To me it seems like `FnvHashMap` is redundant when you're using `NoHashHasher`, but maybe I'm missing something.

Reply:
The `NoHashHasher` is faster and prefferable however `FnvHashMap` supports `TypeId` while the identity-hasher does not.  Since `TypeId` is from my understanding, in current rust, an `u64` we could try to use this however it seems like relying on `TypeId` always being an `u64` is discouraged by the rust community.

Reply:
Oh, I see. I've misread this line of code. I first interpreted it as a `FnvHashMap` with its hash-function set to `NoHashHasher` which is what seemed redundant. But now I see that it's a `FnvHashMap` with its value being a `HashSet` using `NoHashHasher`. Once you've implemented the type aliases this should become much clearer to readers (and me lol)